src/os_posix/os_union_fs.c

```diff
@@ -11,13 +11,14 @@
 
 #define WT_UNION_FS_TOMBSTONE_SUFFIX ".deleted"
 
-static int __union_merge_with_next_extents(WT_SESSION_IMPL *session, WT_UNION_ALLOC_LIST *extent);
 static int __union_fs_file_read(
   WT_FILE_HANDLE *file_handle, WT_SESSION *wt_session, wt_off_t offset, size_t len, void *buf);
 static int __union_fs_file_size(
   WT_FILE_HANDLE *file_handle, WT_SESSION *wt_session, wt_off_t *sizep);
 
-#define EXTENT_END(ext) ((ext)->off + (wt_off_t)(ext)->len)
+/* FIXME - Make these static functions? */
+#define EXTENT_END(ext) ((ext)->off + (wt_off_t)(ext)->len - 1)
+#define ADDR_IN_EXTENT(addr, ext) ((addr) >= (ext)->off && (addr) <= EXTENT_END(ext))
```

Comment:
`ADDR` is too overloaded, just call this `OFFSET_IN_EXTENT` or similar.

Reply:
In terms of `((addr) >= (ext)->off && (addr) <= EXTENT_END(ext))` I think the second part of the `&&` having `<=` is correct as END does a `-1`. Is that right? If so I think we need a comment. Ideally this will all be replaced by the WT_EXTENT migration.

Reply:
Yep, ranges are inclusive so we want the `<=`. I've added a comment in [244bf44](https://github.com/wiredtiger/wiredtiger/pull/11268/commits/244bf444361185894d3c45215bcb0b91e9357af1)